packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -250,18 +251,27 @@ abstract class Domain {
   final Daemon daemon;
   final String name;
   final Map<String, CommandHandler> _handlers = <String, CommandHandler>{};
+  final Map<String, CommandHandlerWithBinary> _handlersWithBinary = <String, CommandHandlerWithBinary>{};
 
   void registerHandler(String name, CommandHandler handler) {
+    assert(!_handlersWithBinary.containsKey(name));
```

Comment:
should this not be:

```
    assert(!_handlers.containsKey(name));
```

Reply:
My intention is to check that the same method name is not registered in both command map.

But I think it doesn't harm to check both maps, and less error prone too. I'll update to check both maps.